File: experiments/014-LiveTypingFiveRuns/cells/5-LiveTyping_5-LiveTyping+8-DesignHeuristicsInline_free/2019-2c-parcial2-masked/20260917-025824-15690/exercise/java/src/main/java/surveycrawler/CrawlerFacingRight.java
package surveycrawler;

public class CrawlerFacingRight extends CrawlerFacing {

    // facing name

    @Override
    public String facingName() {
        return "Right";
    }

    // moving

    @Override
    public void retreat(SurveyCrawler aCrawler) {
        aCrawler.moveLeft();
    }

    @Override
    public void advance(SurveyCrawler aCrawler) {
        aCrawler.moveRight();
    }

    // facing

    @Override
    public void turnCounterClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceUp();
    }

    @Override
    public void turnClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceDown();
    }
}

File: exercises/2019-2c-parcial2-masked/python/src/surveycrawler/crawler_facing_right.py
from surveycrawler.crawler_facing import CrawlerFacing


class CrawlerFacingRight(CrawlerFacing):

    # facing name

    @classmethod
    def facing_name(cls):
        return "Right"

    # moving

    def retreat(self, a_crawler):
        a_crawler.move_left()

    def advance(self, a_crawler):
        a_crawler.move_right()

    # facing

    def turn_counter_clockwise(self, a_crawler):
        a_crawler.face_up()

    def turn_clockwise(self, a_crawler):
        a_crawler.face_down()
